core/DataAccess/ArchiveSelector.php

```diff
@@ -93,16 +103,36 @@ public static function getArchiveIdAndVisits(ArchiveProcessor\Parameters $params
         $visitsConverted = isset($result['nb_visits_converted']) ? $result['nb_visits_converted'] : false;
         $value = isset($result['value']) ? $result['value'] : false;
 
+        $existingRecords = null;
+
         $result['idarchive'] = empty($result['idarchive']) ? [] : [$result['idarchive']];
-        if (isset($result['partial'])) {
-            $result['idarchive'] = array_merge($result['idarchive'], $result['partial']);
+        if (!empty($result['partial'])) {
+            // TODO: comment for this
+            if (!empty($result['idarchive'])
+                || empty($requestedReport)
+            ) {
+                $result['idarchive'] = array_merge($result['idarchive'], $result['partial']);
+            } else {
+                $existingRecords = self::getModel()->getRecordsContainedInArchives($dateStart, $result['partial'], $requestedReport);
```

Comment:
note for myself. this is what the data may look like (the requested record names) in the API.

<img width="726" alt="image" src="https://user-images.githubusercontent.com/273120/225188575-bafd973e-1e67-4c1d-b925-2bab0dffc79c.png">